### PR TITLE
refactor: tidy up getFirstComponentChild

### PR DESCRIPTION
**Pull request by Simon-He95**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `main` branch for v2.x (or to a previous version branch)
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: src/core/vdom/helpers/get-first-component-child.ts [hunk 1]
```diff
@@ -5,12 +5,7 @@ import { isAsyncPlaceholder } from './is-async-placeholder'
...
-  if (isArray(children)) {
-    for (let i = 0; i < children.length; i++) {
-      const c = children[i]
-      if (isDef(c) && (isDef(c.componentOptions) || isAsyncPlaceholder(c))) {
-        return c
-      }
-    }
-  }
+  return isArray(children)
+    ? children.find(c => isDef(c) && (isDef(c.componentOptions) || isAsyncPlaceholder(c)))
```
Comment (sxzz) [inline src/core/vdom/helpers/get-first-component-child.ts:9]: `Array.prototype.find` is not supported in IE.

**Comment by Justineo** [comment]
As @sxzz said, this change will break our current browser compatibility.

**Comment by Simon-He95** [comment]
proxy is still not supported in IE

**Comment by Simon-He95** [comment]
image

in compileScript has used find

**Comment by Justineo** [comment]
The code you proposed to change is runtime code.

**Comment by Simon-He95** [comment]
> The code you proposed to change is runtime code.

Thanks for your explanation

**Pull request closed without merging**